### PR TITLE
Complete rounded rectangle for software draw

### LINKED ISSUE #3547: Implement rounded rectangle in the software painter along with rendering tests
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

The new support for rounded rectangle is targeting GL and GL ES. It does not cover the software painter and some work are needed there to cover this new features.

### Is it possible to construct a solution with the existing API?

No.

### Describe the solution you'd like to see.

Implement rounded rectangle in the software painter.

### PR DESCRIPTION
**Pull request by andydotxyz**
Adding rounded support to software rendering.

Also changed the field from `Radius` to `CornerRadius` as was suggested for clarity.

Fixes #3547

### Checklist:

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.
- [x] Any breaking changes have a deprecation path or have been discussed.

### CODE REVIEW

**Comment by andydotxyz** [comment]
Hopefully @renlite agrees with the change to name. It seems to read better in usage (sorry for not suggesting it earlier).

File: internal/painter/draw.go [hunk 1]
```diff
@@ -98,18 +98,39 @@ func DrawRectangle(rect *canvas.Rectangle, vectorPad float32, scale func(float32
...
+		} else {
+			r := float64(scale(rect.CornerRadius))
+			rasterx.AddRoundRect(float64(p1x), float64(p1y), float64(p3x), float64(p3y), r, r, 0, rasterx.RoundGap, filler)
+		}
 		filler.Draw()
 	}
 
 	if rect.StrokeColor != nil && rect.StrokeWidth > 0 {
+		r := scale(rect.CornerRadius)
+		c := (1 - 0.55228) * r
```
Comment (Jacalz) [inline internal/painter/draw.go:112]: Where does this `0.55228` value come from?

**Review by Jacalz — COMMENTED** [review]
Looking good. Just a few questions.

File: internal/painter/draw.go (hunk 1 above)
Comment (andydotxyz) [reply internal/painter/draw.go]: Sorry it was one of those sneaky mathematical constants in Bézier curves. I've tried to tidy it up.

**Review by Bluebugs — APPROVED** [review]

**Review by Jacalz — APPROVED** [review]

**Pull request merged**